### PR TITLE
Mirascope + Anthropic + Streamlit Website URL Extraction Demo

### PR DESCRIPTION
**Pull request by willbakst**
- Uses Mirascope `AnthropicExtractor` to generate a schema based on a user query.
- Uses another `AnthropicExtractor` to extract the generated schema from scraped data from the given url.
- Streamlit app for easy use and deployment with Ploomber.
- README blog post

### CODE REVIEW

**Comment by edublancas** [comment]
@bryannho please review this

this is a demo + blog post, please review both. once that's done please open a PR to our website repo so we publish the blog post and link to this example

**Comment by edublancas** [comment]
@neelasha23 please also review

**Review by bryannho — APPROVED** [review]
@willbakst Looks great! Drafted a blog post which we'll publish pending @neelasha23's review.

@edublancas Do we want to deploy a live demo as well?

File: examples/streamlit/mirascope-url-extractor/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+
+# Extracting Structured Data from Websites with Mirascope, Anthropic, and Streamlit
+
+## Introduction
+
+Extracting structured data from websites can be a tedious and time-consuming task. Often, the information we need is buried in unstructured HTML pages, requiring us to write custom scrapers or rely on brittle XPath selectors. But what if we could simply describe the data we want in plain English and have an AI-powered tool extract it for us?
+
+That's exactly what the Mirascope URL Extractor achieves. By leveraging the Mirascope library, Anthropic's natural language models, and the Streamlit framework, it allows users to extract structured data from webpages using simple natural language queries. Let's walk through how it works step-by-step.
```
Comment (neelasha23) [inline examples/streamlit/mirascope-url-extractor/README.md:8]: Please add a link to the Github/ docs page: [Mirascope URL Extractor](url)

**Comment by neelasha23** [comment]
A few observations:

1. Please add the `requirements.lock.txt` file
2. Since there is a corresponding PR to blog post, we should remove the blog content from README and include more details on setting up the app locally. Need to add the API keys that should be added as env variables. Also, need to add the URL for generating Anthropic key (https://console.anthropic.com/settings/keys ?)
3. I think we need credits for running it so I haven't been able to test the functionality
Screenshot 2024-04-09 at 12 59 04 PM

@willbakst @edublancas

**Comment by edublancas** [comment]
shared an anthropic key via slack

**Comment by edublancas** [comment]
> Since there is a corresponding PR to blog post, we should remove the blog content from README 

I think let's keep the post (let's just renamed it to post.md and have a readme.md with a shorter description and essential details). once all observations are addressed and this is merged, we can move this to our website's repo (link from the blog post to this repo), and delete it from here

File: examples/streamlit/mirascope-url-extractor/README.md (hunk 1 above)
Comment (willbakst) [reply examples/streamlit/mirascope-url-extractor/README.md]: Done.

**Comment by willbakst** [comment]
> A few observations:
> 
> 1. Please add the `requirements.lock.txt` file
> 2. Since there is a corresponding PR to blog post, we should remove the blog content from README and include more details on setting up the app locally. Need to add the API keys that should be added as env variables. Also, need to add the URL for generating Anthropic key (https://console.anthropic.com/settings/keys ?)
> 3. I think we need credits for running it so I haven't been able to test the functionality
> 
> Screenshot 2024-04-09 at 12 59 04 PM
> @willbakst @edublancas

1. Done
2. I've renamed the post to `post.md` as Eduardo suggested and added a true README.md
3. Yes, you need an API key with credits.

**Comment by neelasha23** [comment]
Works fine but I think we need to refresh for every query? The first time I tried a new query it displayed the previous query's answers and then an error. but on refreshing the page it shows the correct response.

First attempt (Without refresh). Note that it still displays the painting title and links output of the default query

Screenshot 2024-04-09 at 8 57 04 PM

Second attempt:

Screenshot 2024-04-09 at 8 58 17 PM

@willbakst @edublancas

**Comment by willbakst** [comment]
@neelasha23 I've just pushed some changes to hopefully fix some concerns:
1. Your first attempt seems to have hit the token/min limit (which is why it failed).
2. I've updated the streamlit app so if you change the url or query it should clear the previous output. It will also clear when you hit Extract

**Review by neelasha23 — APPROVED** [review]

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
@willbakst great work! I'll follow up with the team so we publish this in our blog, will send you the link once it's live!
